tests/e2e-instrumentation/instrumentation-select/01-assert-select-multi.yaml

```diff
@@ -0,0 +1,77 @@
+apiVersion: v1
+kind: Pod
+metadata:
+  annotations:
+    instrumentation.opentelemetry.io/inject-java: "true"
```

Comment:
I don't understand the use-case to have the `Selector` in the CR while the inject annotation is still needed. 

What is the use-case this feature solves? Why do we need to maintain two approaches to solve the same use-case?

Reply:
This test case is to test that when there are multiple `Instrumentation` (they all use `Selector`), the `Selector` feature can still work as expected.